core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -246,6 +249,15 @@ class HadoopRDD[K, V](
         } catch {
           case eof: EOFException =>
             finished = true
+          case e: Exception =>
```

Comment:
Yes, but it calls into question when you would turn it on. You can't actually handle _just_ the situation you describe reliably even with `IOException`. I think this is band-aiding over an input problem that just isn't properly handled two more steps down the pipeline.